webdriver/tests/classic/perform_actions/pointer_touch.py

```diff
@@ -101,13 +101,31 @@ def test_touch_pointer_properties(session, test_actions_pointer_page, touch_chai
     assert round(events[3]["pressure"], 2) == 0.91
 
 
+def test_touch_pointer_properties_angle_twist(session, test_actions_pointer_page, touch_chain):
+    pointerArea = session.find.css("#pointerArea", all=False)
+    touch_chain.pointer_move(0, 0, origin=pointerArea) \
+        .pointer_down(width=23, height=31, pressure=0.78, altitude_angle=1.2, azimuth_angle=6, twist=355) \
+        .pointer_move(10, 10, origin=pointerArea, width=39, height=35, pressure=0.91, altitude_angle=0.5, azimuth_angle=1.8, twist=345) \
+        .pointer_up() \
+        .pointer_move(80, 50, origin=pointerArea) \
+        .perform()
+    events = get_events(session)
+    assert len(events) == 7
+    event_types = [e["type"] for e in events]
+    assert ["pointerover", "pointerenter", "pointerdown", "pointermove",
+            "pointerup", "pointerout", "pointerleave"] == event_types
+    assert events[2]["type"] == "pointerdown"
+    assert events[2]["tiltX"] == 20
```

Comment:
This expected value seems to be not correct given that this check fails in all browsers. Please see the related wpt.fyi results. The actual value that we get is `0`. Is that correct?

Reply:
It's not correct since the BiDi test passes. Should we just make the changes for BiDi?

Reply:
At least [Firefox](https://wpt.fyi/results/webdriver/tests/bidi/input/perform_actions/pointer_touch.py?diff&filter=ADC&run_id=5161150049681408&run_id=5116221369876480) fails as well here for BiDi. Given that we have the exact same implementation for both protocols this is no wonder. Could there be an issue in Chrome with it's BiDi implementation?

Reply:
Our BiDi implementation is correct. I believe our webdriver implementation is wrong then as well as the Firefox implementation on both.

You can calculate the tilt using the following:

```
function spherical2tilt(altitudeAngle, azimuthAngle) {
  const radToDeg = 180/Math.PI;

  let tiltXrad = 0;
  let tiltYrad = 0;

  if (altitudeAngle == 0) {
    // the pen is in the X-Y plane
    if (azimuthAngle == 0 || azimuthAngle == 2*Math.PI) {
      // pen is on positive X axis
      tiltXrad = Math.PI/2;
    }
    if (azimuthAngle == Math.PI/2) {
      // pen is on positive Y axis
      tiltYrad = Math.PI/2;
    }
    if (azimuthAngle == Math.PI) {
      // pen is on negative X axis
      tiltXrad = -Math.PI/2;
    }
    if (azimuthAngle == 3*Math.PI/2) {
      // pen is on negative Y axis
      tiltYrad = -Math.PI/2;
    }
    if (azimuthAngle>0 && azimuthAngle<Math.PI/2) {
      tiltXrad = Math.PI/2;
      tiltYrad = Math.PI/2;
    }
    if (azimuthAngle>Math.PI/2 && azimuthAngle<Math.PI) {
      tiltXrad = -Math.PI/2;
      tiltYrad = Math.PI/2;
    }
    if (azimuthAngle>Math.PI && azimuthAngle<3*Math.PI/2) {
      tiltXrad = -Math.PI/2;
      tiltYrad = -Math.PI/2;
    }
    if (azimuthAngle>3*Math.PI/2 && azimuthAngle<2*Math.PI) {
      tiltXrad = Math.PI/2;
      tiltYrad = -Math.PI/2;
    }
  }

  if (altitudeAngle != 0) {
    const tanAlt = Math.tan(altitudeAngle);

    tiltXrad = Math.atan(Math.cos(azimuthAngle) / tanAlt);
    tiltYrad = Math.atan(Math.sin(azimuthAngle) / tanAlt);
  }

  return {"tiltX":tiltXrad*radToDeg, "tiltY":tiltYrad*radToDeg};
}
```

Reply:
`spherical2tilt(1.2, 6) == {tiltX: 20.47033124113985, tiltY: -6.199788932725595}`

Reply:
I would really like to know / understand what's going on here. @jrandolf where is this calculation algorithm coming from?

@sadym-chromium do you have an idea why for Chrome the behavior is different between WebDriver [classic](https://wpt.fyi/results/webdriver/tests/classic/perform_actions/pointer_touch.py?diff&filter=ADC&run_id=5153511383236608&run_id=6278527063687168) and [BiDi](https://wpt.fyi/results/webdriver/tests/bidi/input/perform_actions/pointer_touch.py?diff&filter=ADC&run_id=5153511383236608&run_id=6278527063687168)? The classic behavior also matches the value from other browsers.

Reply:
Sorry, forgot to link the algorithm: https://www.w3.org/TR/pointerevents3/#converting-between-tiltx-tilty-and-altitudeangle-azimuthangle

Reply:
Thanks! I had another looks and I think that I found the problem. It's indeed related to our WebDriver implementation. We [miss to actually correctly set both properties](https://searchfox.org/mozilla-central/rev/e94bcd536a2a4caad0597d1b2d624342e6a389c4/remote/shared/webdriver/Actions.sys.mjs#2242-2253). As such it's clear that values will be wrong.

We should get those changes landed and I'll file a bug on our side to get the behavior corrected.

Note that this still doesn't answer the question why chromedriver behaves differently.

Reply:
From looking at our code, [it looks like we also do not set them!](https://source.chromium.org/chromium/chromium/src/+/main:chrome/test/chromedriver/window_commands.cc;l=357;drc=7fa0c25da15ae39bbd2fd720832ec4df4fee705a;bpv=0;bpt=1)

Reply:
Ah! Are you going to file a bug for chromedriver or should I do it?

Reply:
Actually I filed https://bugs.chromium.org/p/chromedriver/issues/detail?id=4623 now.